core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -427,20 +427,46 @@ default void setResponseTimeout(Duration responseTimeout) {
         setResponseTimeout(TimeoutMode.SET_FROM_NOW, responseTimeout);
     }
 
+    /**
+     * Returns a {@link CompletableFuture} which is completed with a {@link Throwable} cancellation cause when
+     * {@link ClientRequestContext} is about to get cancelled.
+     */
+    CompletableFuture<Throwable> whenResponseCancelling();
+
+    /**
+     * Returns a {@link CompletableFuture} which is completed with a {@link Throwable} cancellation cause after
+     * {@link ClientRequestContext} has been cancelled. {@link #isCancelled()} will always return {@code true}
+     * when the returned {@link CompletableFuture} is completed.
+     */
+    CompletableFuture<Throwable> whenResponseCancelled();
```

Comment:
How about renaming these methods to `whenCancelling()` and `whenCancelled()` and pulling them up to `RequestContext`?